### PR TITLE
AL-524 - Logging Service

### PR DESCRIPTION
**Pull request by mjoksa**

### CODE REVIEW

File: src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Services/UserLogService.cs [hunk 1]
```diff
@@ -11,28 +12,92 @@ namespace Lykke.AlgoStore.CSharp.AlgoTemplate.Services.Services
...
+
+            _actionBlock = new ActionBlock<UserLogRequest[]>(x => PersistUserLogs(x),
+                new ExecutionDataflowBlockOptions {MaxDegreeOfParallelism = 2});
+
+            _batchBlock.LinkTo(_actionBlock);
+
+            ExtendBatchExpiration();
+
+            //Initiate timer after 1s (1000ms) with a period of 10s (10000ms)
+            _timer = new Timer(HandleTimerTriggered, null, 1000, 10000);
```
Comment (dafinkagerdjikova) [inline src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Services/UserLogService.cs]: Why do we need this timer?

Comment (mjoksa) [reply src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Services/UserLogService.cs]: Timer is used to make sure that logs will be saved in timely fashion even when log batch is not full

Comment (ghost) [reply src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Services/UserLogService.cs]: What happens if the max batch lifetime is shorter than 10s?

Comment (mjoksa) [reply src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Services/UserLogService.cs]: I've changed timer period to 50ms

**Review by ghost — APPROVED** [review]

**Review by dafinkagerdjikova — APPROVED** [review]

**Pull request merged**